src/pages/index.js

```diff
@@ -7,9 +7,51 @@ import Section from "../components/Section.js";
 import FormValidator from "../components/FormValidator.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
+import PopupConfirmation from "../components/PopupConfirmation.js";
 import UserInfo from "../components/UserInfo.js";
 import "./index.css";
 import Constants from "../utils/constants.js";
+import Api from "../components/Api.js";
+
+// ! ||--------------------------------------------------------------------------------||
+// ! ||                                      API                                       ||
+// ! ||--------------------------------------------------------------------------------||
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "7d9b26b0-78d0-422a-941a-09c95ea2b082",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getUserInfo()
+  .then((userData) => {
+    profileInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+    });
+    profileInfo.updateAvatar(userData.avatar);
+  })
+  .catch(api.processError);
+
+function handleUpdateProfile(userData) {
+  api
+    .updateProfile(userData)
+    .then((userData) => {
+      profileInfo.setUserInfo({
+        name: userData.name,
+        description: userData.about,
+      });
+    })
+    .catch(api.processError);
+}
+
+function handleUpdateAvatar(userData) {
+  api.updateAvatar(userData);
```

Comment:
Needs correction: don't forget about showing the loading state on the button and handling the api errors